requirements.txt

```diff
@@ -0,0 +1,13 @@
+# runtime
```

Comment:
What's this file for?

Reply:
The setup.py and requirements.txt do different things. The setup.py is package metadata used in the installation of the package: it contains hard runtime dependencies, and optional runtime dependencies under "extras", with loose-as-possible version constraints (if it's a library or has to coexist with other python packages). The requirements.txt is for developers and testers of the package: it contains all possible runtime dependencies, ideally with fixed versions to test against, and all test, doc, format, and build dependencies.

Without a requirements.txt, someone interested in contributing has to dig through the setup.py (and hope the extras are all well-described), all of the CI config, the Makefile, and potentially all of the test files to find what they actually need to install to make sure it works. With a requirements.txt, it's `git clone...; pip install -r requirements.txt` and they're away.

It doesn't have to stay, it's just something I would have found useful a couple of hours ago.

Reply:
It can also help shrink CI configs as all of the dependencies can be listed in one place.

Reply:
I see where you're coming from. It's just that now, if one adds or removes a dependency one has to do it in two files, and I can already see how this gets messed up. Let's keep it for now, but I might remove it in the future.

Reply:
Agreed, it feels counterintuitive. I haven't had any issues with this pattern myself, but the possibility is there.